Javadoc parsing error with multi-line text and @throws

```
import java.io.IOException;

public class ActionTestHelper {

    /**
     * Delete the stop file if it exists.
     * This is necessary if the stop file was not deleted during the test.
     *
     * @throws IOException if the stop file exists, but could not be deleted.
     */
    public static void cleanupStopFile() throws IOException {
    }
}
```

I checked out the current `main` branch once more, and extended the exception with more details:

        int endIndex = source.indexOf(delim, cursor);
        if (endIndex < 0) {
            throw new IllegalStateException("Expected to be able to find " + delim + " in \"" +
                    source.substring(cursor).replace("\\", "\\\\").replace("\n", "\\n").replace("\r", "\\r").replace("\"", "\\\"") + "\"");
        }

I published to maven local and ran this command on a dummy project containing only the file above (with no-daemon because otherwise I'd need to stop the daemon each time I publish the rewrite libraries to maven local):

    gradlew rewriteDryRun --no-daemon --stacktrace -DactiveRecipe=org.openrewrite.java.RemoveUnusedImports

I got this exception for the file above formatted with Unix linebreaks:

    java.lang.IllegalStateException: Expected to be able to find @exception in "he test.\r     *\r     * @throws IOException if the stop file exists, but could not be deleted.\r     "

With Windows linebreaks, I got this exception instead:

    java.lang.IllegalStateException: Expected to be able to find @exception in ".\r\n\r\n @throws IOException if the stop file exists, but could not be deleted.\r"

Seems that `Java11JavadocVisitor.visitThrows` expects the `cursor` to be at the `@throws`, but it is not, so it starts looking for `@exception` instead of `@throws`.

For reference, the original exception, as of release 7.13.0, is this:

    Caused by: java.lang.IllegalStateException: Expected to be able to find @exception
        at org.openrewrite.java.Java11JavadocVisitor.sourceBefore(Java11JavadocVisitor.java:859)
        at org.openrewrite.java.Java11JavadocVisitor.visitThrows(Java11JavadocVisitor.java:781)
        at org.openrewrite.java.Java11JavadocVisitor.visitThrows(Java11JavadocVisitor.java:46)
        at jdk.compiler/com.sun.tools.javac.tree.DCTree$DCThrows.accept(DCTree.java:932)
        at jdk.compiler/com.sun.source.util.DocTreeScanner.scan(DocTreeScanner.java:79)
        at org.openrewrite.java.Java11JavadocVisitor.visitDocComment(Java11JavadocVisitor.java:305)
        at org.openrewrite.java.Java11JavadocVisitor.visitDocComment(Java11JavadocVisitor.java:46)
        at jdk.compiler/com.sun.tools.javac.tree.DCTree$DCDocComment.accept(DCTree.java:138)
        at jdk.compiler/com.sun.source.util.DocTreeScanner.scan(DocTreeScanner.java:79)
        at org.openrewrite.java.Java11ParserVisitor.formatWithCommentTree(Java11ParserVisitor.java:1936)
        at org.openrewrite.java.Java11ParserVisitor.convert(Java11ParserVisitor.java:1434)

--- a/rewrite-java-11/src/main/java/org/openrewrite/java/Java11JavadocVisitor.java
+++ b/rewrite-java-11/src/main/java/org/openrewrite/java/Java11JavadocVisitor.java
@@ -122,9 +122,6 @@ public class Java11JavadocVisitor extends DocTreeScanner<Tree, List<Javadoc>> {
                         String prevNewLine = i > 1 && sourceArr[i - 2] == '\r' ? "\r\n" : "\n";
                         lineBreaks.put(javadocContent.length(), new Javadoc.LineBreak(randomId(), prevNewLine, Markers.EMPTY));
                     }
-                    if (sourceArr[i - 1] == '\r') {
-                        javadocContent.append('\r');
-                    }
                     javadocContent.append(c);
                 }
                 String newLine = sourceArr[i - 1] == '\r' ? "\r\n" : "\n";

--- a/rewrite-java-8/src/main/java/org/openrewrite/java/ReloadableJava8JavadocVisitor.java
+++ b/rewrite-java-8/src/main/java/org/openrewrite/java/ReloadableJava8JavadocVisitor.java
@@ -122,9 +122,6 @@ public class ReloadableJava8JavadocVisitor extends DocTreeScanner<Tree, List<Jav
                         String prevNewLine = i > 1 && sourceArr[i - 2] == '\r' ? "\r\n" : "\n";
                         lineBreaks.put(javadocContent.length(), new Javadoc.LineBreak(randomId(), prevNewLine, Markers.EMPTY));
                     }
-                    if (sourceArr[i - 1] == '\r') {
-                        javadocContent.append('\r');
-                    }
                     javadocContent.append(c);
                 }
                 String newLine = sourceArr[i - 1] == '\r' ? "\r\n" : "\n";

--- a/rewrite-test/src/main/kotlin/org/openrewrite/java/tree/JavadocTest.kt
+++ b/rewrite-test/src/main/kotlin/org/openrewrite/java/tree/JavadocTest.kt
@@ -1008,8 +1008,8 @@ interface JavadocTest : JavaTreeTest {
     @Test
     fun javaDocWithCRLF(jp: JavaParser) = assertParsePrintAndProcess(
         jp,
-        CompilationUnit,
-        "/**\r\n" +
+        CompilationUnit, "" +
+              "/**\r\n" +
               " * JavaDoc.\r\n" +
               " */\r\n" +
               "public class A {\r\n" +
@@ -1046,5 +1046,40 @@ interface JavadocTest : JavaTreeTest {
             class Test {
             }
         """.trimIndent()
+    )
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/976")
+    @Test
+    fun multilineJavaDocWithCRLF(jp: JavaParser) = assertParsePrintAndProcess(
+        jp,
+        CompilationUnit, "" +
+                "/**\r\n" +
+                " * Line 1.\r\n" +
+                " * Line 2.\r\n" +
+                " */\r\n" +
+                "public class A {\r\n" +
+                "    /**\r\n" +
+                "     * Line 1.\r\n" +
+                "     * Line 2.\r\n" +
+                "     */\r\n" +
+                "    void method() {}\r\n" +
+                "}"
+    )
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/976")
+    @Test
+    fun multilineWithThrowsAndCRLF(jp: JavaParser) = assertParsePrintAndProcess(
+        jp,
+        CompilationUnit, "" +
+                "import java.io.IOException;\r\n" +
+                "\r\n" +
+                "public class A {\r\n" +
+                "    /**\r\n" +
+                "     * Line 1.\r\n" +
+                "     * Line 2.\r\n" +
+                "     * @throws IOException text.\r\n" +
+                "     */\r\n" +
+                "    void method() throws IOException {}\r\n" +
+                "}"
     )
 }

